## Final Project Submission

Please fill out:
* Student name: Pacificah Asamba
* Student pace: self paced / part time / full time - part time
* Scheduled project review date/time:  
* Instructor name: Madam Mwikali
* Blog post URL:


# 1. Import the relevant libraries 

In [9]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

## 2. Load the Data into a DataFrame Called `Aviation_data`

The file path is `data/Aviation_Data.csv`. Use pandas to read in the data from this CSV file and create a dataframe named `Aviation_data`.

In [10]:
#load the Aviation data csv file using pandas to a dataframe called Aviation_Data

Aviation_data = pd.read_csv('data/Aviation_Data.csv')
Aviation_data.head()   #run the .head() to view a sample of how our data looks like

c:\Users\Admin\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (6,7,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   
3       EUREKA, CA  United States      NaN       NaN          NaN   
4       Canton, OH  United States      NaN       NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

# 3. Data understanding

### Now that we have read our data to a dataframe, we are going to get to understand our data. This is done by using various methods as below:

* 1. First we need to know the shape of the data by using the .shape function
* 2. Next we will check the various datatypes we have, the number of rows and columns and which columns have null values using the .info() method
* 3. To know the descriptive statistics we will use the .describe method

In [11]:
# To know the number of rows and columns, do a .shape
Aviation_data.shape

(90348, 31)

* From the above the dataset has 90,348 rows and 31 columns

In [12]:
#check the names of the columns
Aviation_data.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [13]:
#using the .info()method for more data understanding
Aviation_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

# info() method output analysis
1. The data is of class - pandas.core.frame.DataFrame
2. There are 90,348 entries(rows) from index 0 to 90,347
3. There are 31 columns
4. All columns except Investgation.Type have missing values
5. Twenty six columns have data of type 'object' and five columns have type float64

In [ ]:
#This gives the summary statistics for the data set for the numeric columns
Aviation_data.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [15]:
# Your code here - remember to use markdown cells for comments as well!

In [17]:
Aviation_data.isna().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40099
Airport.Name              37558
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2776
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8536
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7840
Publication.Date          16689
dtype: i

In [26]:
Aviation_data.sample(5)

Event.Id Investigation.Type Accident.Number  Event.Date  \
21532  20001213X26895           Accident      MKC88LA177  1988-09-25   
41695  20001208X07898           Accident      FTW97FA174  1997-05-03   
75167  20140409X60140           Accident      CEN14LA192  2014-04-09   
61343  20061006X01483           Incident      NYC06IA226  2006-09-01   
20934  20001213X26400           Accident      SEA88LA137  1988-07-23   

                  Location        Country   Latitude   Longitude Airport.Code  \
21532            OMAHA, NE  United States        NaN         NaN          OMA   
41695      LAKE GEORGE, CO  United States        NaN         NaN         CO96   
75167      Littlefield, TX  United States    335437N    1021949W          LIU   
61343  Matinicus Islnd, ME  United States  43.873334  -68.894722         35ME   
20934    COEUR D'ALENE, ID  United States        NaN         NaN          NaN   

                        Airport.Name  ... Purpose.of.flight  \
21532                         EPPLEY  ...          Personal   
41695                        PERSONS  ...          Personal   
75167  Littlefield Taylor Brown Muni  ...          Personal   
61343       Matinicus Island Airport  ...               NaN   
20934                            NaN  ...          Personal   

                                          Air.carrier Total.Fatal.Injuries  \
21532                                             NaN                  0.0   
41695                                             NaN                  2.0   
75167                                             NaN                  1.0   
61343  Waters Aero-marine (dba: Penobscot Island Air)                  NaN   
20934                                             NaN                  0.0   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
21532                    0.0                  0.0             4.0   
41695                    0.0                  0.0             0.0   
75167                    0.0                  0.0             0.0   
61343                    NaN                  NaN             3.0   
20934                    0.0                  2.0             0.0   

      Weather.Condition  Broad.phase.of.flight  \
21532               VMC               Approach   
41695               VMC               Approach   
75167               VMC                    NaN   
61343               VMC               Standing   
20934               VMC                  Climb   

                                           Report.Status Publication.Date  
21532                                     Probable Cause       12-07-1989  
41695                                     Probable Cause       10-04-1998  
75167  The non-certificated pilot's in-flight loss of...       25-09-2020  
61343                                     Probable Cause       28-02-2008  
20934                                     Probable Cause       07-06-1989  

[5 rows x 31 columns]

# 4. Data cleaning
Now that we have loaded our data and understood it by applying some dataframe methods, we need to do some data cleaning. In this section I will be focussing on:
i. Those columns which have missing values
ii. Columns with invalid values
iii. Check for duplicates
iv. 

